api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -146,19 +205,21 @@ private Order saveOrder(Order order, OrderContext orderContext, boolean isRetros
 		if (isDrugOrder) {
```

Comment:
extract this if block into another method called

`ensureDrugOrderAutoExpirationDateIsSet`

reason is this: the method we extracted is `ensureConceptIsSet` but this block takes care of something else so keeping this in here is misleading of what the method is doing. Therefore extract it and with the new helper `isDrugOrder(order)` thats an easy task :)

call this method then in the saveOrder right after ensureConceptIsSet like so

```java
		failOnExistingOrder(order);
		ensureDateActivatedIsSet(order);
		ensureConceptIsSet(order);
		ensureDrugOrderAutoExpirationDateIsSet(order);
		ensureOrderTypeIsSet(order,orderContext);
		ensureCareSettingIsSet(order,orderContext);
		failOnOrderTypeMismatch(order);
```